aws_xray_sdk/ext/pymongo/patch.py

```diff
@@ -0,0 +1,36 @@
+from pymongo import monitoring
+from aws_xray_sdk.core import xray_recorder
+
+
+class XrayCommandListener(monitoring.CommandListener):
+    """
+    A listener that traces all pymongo db commands to AWS Xray.
+    Creates a subsegment for each mongo db conmmand.
+
+    name: 'MongoDB-127.0.0.1:27017'
+    annotations: command_name:str, database_name: str, succeed: bool
+    """
+
+    def started(self, event):
+        host, port = event.connection_id
+        subsegment = xray_recorder.begin_subsegment(
+            f'MongoDB-{host}:{port}', 'remote')
+        subsegment.put_annotation('command_name', event.command_name)
```

Comment:
X-Ray service aggregates annotations at trace level when indexing. If another backend DB has a subsegment which has the same key, one will be overridden which is not what we want. A prefix `mongodb` is probably a better option.

Reply:
What happens when a segment has multiple pymongo calls?